lib/sqlalchemy/orm/mapped_collection.py

```diff
@@ -381,12 +436,14 @@ def remove(self, value, _sa_initiator=None):
                 "based on mutable properties or properties that only obtain "
                 "values after flush?" % (value, self[key], key)
             )
-        self.__delitem__(key, _sa_initiator)
+        self.__delitem__(key, _sa_initiator)  # type: ignore[call-arg]
 
 
-def _mapped_collection_cls(keyfunc, ignore_unpopulated_attribute):
-    class _MKeyfuncMapped(KeyFuncDict):
-        def __init__(self):
+def _mapped_collection_cls(
+    keyfunc: _AnyCallable, ignore_unpopulated_attribute: bool
+) -> type[KeyFuncDict[_KT, _KT]]:
```

Comment:
for consistency since we have things like `List[]` this should be `Type[]` until we are fully python 3.10 in a few years

Reply:
Yes, this is my fault, I've fixed that in latest commit